node/docs/nodeVersioning.md

```diff
@@ -0,0 +1,20 @@
+# Node versioning
```

Comment:
i left some comments of things to think through. otherwise looks good to me

Reply:
I responded to everything, could probably use another review before merging.

Reply:
I left a couple small comments. Looks good to me. Nice work.